### DISCUSSION TITLE
Is onProgress supposed to run while the screen is locked (during background play)?

### DISCUSSION THREAD

**Discussion by arapocket** [TheWidlarzGroup/react-native-video#4267] [Q&A] [answered]
### What are you trying to do?

Determine whether `onProgress` is supposed to run while the screen is locked (during background play). My experience is that it stops after a minute or so.

### What have you tried?

Tested with background play set up, which works fine. The `onProgress` callback stops firing after a minute or so while the screen is locked. Tested on both Android and iOS.

### Platforms

iOS, Android

**Comment by freeboub** [comment]
Yes, if playback is ongoing it should be sent. But as react stops rendering when the app is stopped , it is possible that your app doesn't receive it in that case... But is it really needed ?

**Reply by arapocket** [reply]
For more context, my use case is tracking a user's progress on a video (during airplay) while their screen is locked. I need to be making api requests at a set interval as the video plays.

I understand this is probably out of the scope of this library.

**Reply by arapocket** [reply]
On further testing I found it keeps running as long as the screen isn't locked. So if for example i've minimized the app, it continues indefinitely.

**Reply by freeboub** [reply]
Maybe you should check around power management 🤷 I am an ios dev, but on android there are some walkelock which can be taken to avoid cpu to go into low power mode. (If that is your issue)

**Comment by arapocket** [comment, marked as answer]
For anyone that sees this and is curious, looks like the real issue is that my api calls aren't going through when the screen is locked. onProgress is continuing to be called (at least when my device is hooked up to the metro bundler). 

edit: Confirmed, works in a prod build as well
